TACC/Core-CMS-Custom
# fix(apcd-cms): fp-1837 superuser must see all menu items

## Overview

Do **not** restrict **superuser** ability to see menu items.

## Related

- [FP-1837](https://jira.tacc.utexas.edu/browse/FP-1837)
- fixes https://github.com/TACC/Core-CMS-Custom/pull/45

## Changes

- make `limit_visibility_in_menu` never limit a superuser

## Testing

0. Check out this PR's branch.
1. Re-build and re-up containers.
    ```
    docker-compose -f docker-compose.dev.yml down && docker-compose -f docker-compose.dev.yml build --no-cache && 
    docker-compose -f docker-compose.dev.yml up --force-recreate --detach
    ```
2. Login to CMS admin (http://localhost:8000/admin) as a superuser.
3. Create user groups:
    - `SUBMITTER_USER`
    - `SUBMITTER_ADMIN`
    - `APCD_ADMIN`
4. Create three pages that are visible in menu:
    - Home
    - Submissions (Advanced Settings: ID: `submissions`)
    - Administration (Advanced Settings: ID: `administration`)
5. (if you had tested [#45](https://github.com/TACC/Core-CMS-Custom/pull/45)) Ensure your CMS superuser does **not** belong to _any_ group.
6. Load any CMS webpage (not the `/admin` interface).
7. Verify these items are in the CMS nav:
    - Home
    - Submissions
    - Administration

## UI

| Edit Superuser<br /><sup>(no groups)</sup> | Load Page as Superuser<br /><sup>(all menu items)</sup>
| - | - |
| ![Django Admin Edit Superuser](https://user-images.githubusercontent.com/62723358/201249650-562584c8-7ad3-4624-8fd1-ecf3a547b72f.png) | ![Django CMS Webpage as Superuser](https://user-images.githubusercontent.com/62723358/201249652-1732f38c-04dc-4332-a61a-9e3e6d074867.png) |
